showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -127,6 +138,92 @@ class ShowkaseProcessor @JvmOverloads constructor(
             }.flatten().mapNotNull { it }.toSet()
     }
 
+    private fun processCustomAnnotation(
+        roundEnvironment: XRoundEnv,
+        annotation: XTypeElement? = null
+    ): Set<ShowkaseMetadata.Component> {
+        val supportedTypes = mutableListOf<String>()
+        if (annotation != null) supportedTypes.add(annotation.qualifiedName)
+        val components = mutableSetOf<ShowkaseMetadata.Component>()
+
+        supportedTypes.map { supportedType ->
+            val annotatedElements = roundEnvironment.getElementsAnnotatedWith(supportedType)
+            annotatedElements.map { annotatedElement ->
+                if (!showkaseValidator.checkElementIsAnnotationClass(annotatedElement)) {
+
+                    showkaseValidator.validateComponentElementOrSkip(
+                        element = annotatedElement,
+                        annotationName = supportedType,
+                    )
+                    components.addAll(
+                        getShowkaseMetadataFromCustomAnnotation(
+                            element = annotatedElement,
+                            showkaseValidator = showkaseValidator,
+                            supportedType.getCustomAnnotationSimpleName(),
+                        ).toSet()
+                    )
+                }
+            }
+        }
+        return components
+    }
+
+    private fun String.getCustomAnnotationSimpleName(): String {
+        return this.split(".").last()
+    }
+
+    private fun processCustomAnnotationFromClasspath(roundEnvironment: XRoundEnv): Set<ShowkaseMetadata.Component> {
+        // In this function we are checking generated classpath for MultiPreview codegen annotations.
+        // We also check the current module if there is any composables that are annotated with the qualified name
+        // from the annotation from classpath. We use the fields from the classpath annotation to build
+        // common data for the ShowkaseMetadata.
+
+        // Supported annotations from classpath
+        val supportedCustomPreview = environment.getTypeElementsFromPackage(CODEGEN_PACKAGE_NAME)
+            .flatMap { it.getEnclosedElements() }.mapNotNull {
+                when (
+                    val annotation = it.getAnnotation(ShowkaseMultiPreviewCodegenMetadata::class)
+                ) {
+                    null -> null
+                    else -> annotation.value
+                }
+            }
+        val components = mutableSetOf<ShowkaseMetadata.Component>()
+
+        supportedCustomPreview
+            .mapIndexed { index: Int, customPreviewMetadata: ShowkaseMultiPreviewCodegenMetadata ->
+                roundEnvironment
+                    .getElementsAnnotatedWith(customPreviewMetadata.supportTypeQualifiedName)
```

Comment:
since we aren't registering this annotation in `getSupportedAnnotationTypes`, how can this work for kapt? 🤔 I think there should be a scenario where this wouldn't work

Reply:
Hm, that is a good point 🤔 It works in the UI tests and in the sample apps for the main module and the sample module.  Maybe it should be tested with a third module? 🤔 Wdyt?

Reply:
Added registration on this to 855cd00, but I must say I'm wondering why this works without it and which case it does not 🤔

Reply:
yeah I didn't understand this part either...

Reply:
Are you sure that we really need to register annotation qualified names for it to work with KAPT? Is there any documentation on that? Do u know? :)